### PR TITLE
Timeout configurable

### PR DESCRIPTION
**Pull request by gerring**
Several classes which act as temporary remote services do not have appropriately configurable timeouts. This change attempts to fix that.

### CODE REVIEW

File: org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/AbstractRemoteService.java [hunk 1]
```diff
@@ -27,6 +29,8 @@
  */
```
Comment (AnthonyHullDiamond) [inline org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/AbstractRemoteService.java]: There is no param T in the code

File: org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/_AbstractRemoteDevice.java [hunk 2]
```diff
@@ -39,12 +39,14 @@ private _AbstractRemoteDevice(URI uri, IEventService eservice) throws EventExcep
 	@SuppressWarnings("unchecked")
 	_AbstractRemoteDevice(DeviceRequest req, URI uri, IEventService eservice) throws EventException, InterruptedException {
 		this(uri, eservice);
+	    logger.info("Setting "+getClass().getSimpleName()+" to timeout " + RemoteServiceFactory.getTime() + " "+RemoteServiceFactory.getTimeUnit());
```
Comment (AnthonyHullDiamond) [inline org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/_AbstractRemoteDevice.java]: Do we need getClass().getSimpleName() here? The logger adds the class name

2017-04-11 15:51:57,199 INFO  org.eclipse.scanning.event.remote._RunnableDeviceService - Setting _RunnableDeviceService to timeout 500 ms

File: org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/_Positioner.java [hunk 3]
```diff
@@ -48,7 +49,9 @@ public void disconnect() throws EventException {
 	
 	public void init()  throws EventException {
 		requester = eservice.createRequestor(uri, EventConstants.POSITIONER_REQUEST_TOPIC, EventConstants.POSITIONER_RESPONSE_TOPIC);
-	    requester.setTimeout(RemoteServiceFactory.getTime(), RemoteServiceFactory.getTimeUnit()); // Useful for debugging testing 
+		long timeout = Long.getLong("org.eclipse.scanning.event.remote.positionerTimeout", 100);
```
Comment (AnthonyHullDiamond) [inline org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/_Positioner.java]: Maybe make the default timeout a defined constant at the top of the class (here & elsewhere)

File: org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/AbstractRemoteService.java [hunk 4]
```diff
@@ -27,6 +29,8 @@
  */
 abstract class AbstractRemoteService implements IDisconnectable, Closeable {
 
+	protected final static Logger logger = LoggerFactory.getLogger(AbstractRemoteService.class);
```
Comment (gerring) [inline org.eclipse.scanning.event/src/org/eclipse/scanning/event/remote/AbstractRemoteService.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by gerring** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by gerring** [comment]
Thanks for the comments, I have responded.

**Pull request merged**